### ISSUE TITLE
Gutenberg: Critical Error on Atomic Sites related to `gutenberg_update_wp_date_settings`

### ISSUE THREAD

**Issue by villanovachile** [Automattic/wp-calypso#84299]
### Quick summary

Some Atomic sites are running into critical errors with Gutenberg 17.0.2. The issue affects both front end and back end, making the site completely inaccessible.

Deactivating Gutenberg resolves the issue.

This is currently being discussed:
p1700217003099279-slack-CBTN58FTJ
p1699438975775509-slack-C02TCEHP3HA

### Steps to reproduce

We are currently investigating and identifying how to reliably reproduce this issue. 

### What you expected to happen

I expect there to be no critical errors.

### What actually happened

There are critical errors.

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

```
Fatal error: Uncaught TypeError: array_values(): Argument #1 ($array) must be of type array, null given in /wordpress/plugins/gutenberg/17.0.2/lib/compat/wordpress-6.4/script-loader.php:38
Stack trace:
#0 /wordpress/plugins/gutenberg/17.0.2/lib/compat/wordpress-6.4/script-loader.php(38): array_values(NULL)
#1 /wordpress/core/6.4.1/wp-includes/class-wp-hook.php(324): gutenberg_update_wp_date_settings(Object(WP_Scripts))
#2 /wordpress/core/6.4.1/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#3 /wordpress/core/6.4.1/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#4 /wordpress/core/6.4.1/wp-includes/class-wp-scripts.php(166): do_action_ref_array('wp_default_scri...', Array)
#5 /wordpress/core/6.4.1/wp-includes/class-wp-scripts.php(149): WP_Scripts->init()
#6 /wordpress/core/6.4.1/wp-includes/functions.wp-scripts.php(24): WP_Scripts->__construct()
#7 /wordpress/core/6.4.1/wp-includes/functions.wp-scripts.php(368): wp_scripts()
#8 /wordpress/core/6.4.1/wp-includes/class-wp-editor.php(860): wp_enqueue_script('editor')
#9 /wordpress/core/6.4.1/wp-includes/class-wp-editor.php(910): _WP_Editors::enqueue_scripts(true)
#10 /wordpress/core/6.4.1/wp-includes/general-template.php(3828): _WP_Editors::enqueue_default_editor()
#11 /srv/htdocs/wp-content/plugins/shortcode-imdb/shortcode-imdb.php(17): wp_enqueue_editor()
#12 /wordpress/core/6.4.1/wp-settings.php(473): include_once('/srv/htdocs/wp-...')
#13 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1336): require('/wordpress/core...')
#14 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1254): WP_CLI\Runner->load_wordpress()
#15 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php(28): WP_CLI\Runner->start()
#16 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/bootstrap.php(78): WP_CLI\Bootstrap\LaunchRunner->process(Object(WP_CLI\Bootstrap\BootstrapState))
#17 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/wp-cli.php(32): WP_CLI\bootstrap()
#18 phar:///usr/local/bin/wp-cli/php/boot-phar.php(11): include('phar:///usr/loc...')
#19 /usr/local/bin/wp-cli(4): include('phar:///usr/loc...')
#20 {main}
  thrown in /wordpress/plugins/gutenberg/17.0.2/lib/compat/wordpress-6.4/script-loader.php on line 38
Error: A fost o eroare critică pe acest site web.Află mai multe despre depanarea WordPress. A fost o eroare critică pe acest site web.
```

**Comment by villanovachile** [comment]
Report: 7327459-zd-a8c

**Comment by ramonjd** [comment]
Might be a wild goose chase but I wonder if the shortcode plugin calling [wp_enqueue_editor](https://github.com/kemalyazici/shortcode-imdb/blob/master/shortcode-imdb.php#L17) is triggering anything. This function also enqueues plugin scripts.

`gutenberg_update_wp_date_settings` in https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.4/script-loader.php uses the following hook: 

`add_action( 'wp_default_scripts', 'gutenberg_update_wp_date_settings' );`


 [The global $wp_locale](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.4/script-loader.php#L38) doesn't appear to be available yet when calling `array_values( $wp_locale->month )`. 

Maybe a priority needs to be set on the hook? I don't know, just reaching here... 😄

**Comment by villanovachile** [comment]
> Might be a wild goose chase but I wonder if the shortcode plugin calling [wp_enqueue_editor](https://github.com/kemalyazici/shortcode-imdb/blob/master/shortcode-imdb.php#L17) is triggering anything.

@ramonjd good call out! Deactivating `shortcode-imdb` resolved the issue and allowed Gutenberg to be reactivated. I'll keep an eye out for other occurrences to see if we can spot some trends here and other plugins that may be calling `wp_enqueue_editor`.

**Comment by ramonjd** [comment]
Thanks for confirming @villanovachile 🙇🏻 

I'll take a look on Monday to see if we need to modify the hook in Gutenberg.

**Comment by ramonjd** [comment]
I can reproduce by installing the imdb shortcode link on WordPress 6.5 and using Gutenberg 17.0 (latest).


The notice is actually warning that `wp_enqueue_script` is being called incorrectly.


```
Notice: Function wp_enqueue_script was called incorrectly. Scripts and styles should not be registered or enqueued until the wp_enqueue_scripts, admin_enqueue_scripts, or login_enqueue_scripts hooks. This notice was triggered by the editor handle. Please see Debugging in WordPress for more information. (This message was added in version 3.3.0.) ...
```


<details>

<summary>Here's the full PHP output:</summary>

```
Notice: Function wp_enqueue_script was called incorrectly. Scripts and styles should not be registered or enqueued until the wp_enqueue_scripts, admin_enqueue_scripts, or login_enqueue_scripts hooks. This notice was triggered by the editor handle. Please see Debugging in WordPress for more information. (This message was added in version 3.3.0.) in /var/www/html/wp-includes/functions.php on line 6031 Warning: Attempt to read property "month" on null in /var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.4/script-loader.php on line 38 Fatal error: Uncaught TypeError: array_values(): Argument #1 ($array) must be of type array, null given in /var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.4/script-loader.php:38 Stack trace: #0 /var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.4/script-loader.php(38): array_values(NULL) #1 /var/www/html/wp-includes/class-wp-hook.php(324): gutenberg_update_wp_date_settings(Object(WP_Scripts)) #2 /var/www/html/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array) #3 /var/www/html/wp-includes/plugin.php(565): WP_Hook->do_action(Array) #4 /var/www/html/wp-includes/class-wp-scripts.php(166): do_action_ref_array('wp_default_scri...', Array) #5 /var/www/html/wp-includes/class-wp-scripts.php(149): WP_Scripts->init() #6 /var/www/html/wp-includes/functions.wp-scripts.php(24): WP_Scripts->__construct() #7 /var/www/html/wp-includes/functions.wp-scripts.php(368): wp_scripts() #8 /var/www/html/wp-includes/class-wp-editor.php(860): wp_enqueue_script('editor') #9 /var/www/html/wp-includes/class-wp-editor.php(910): _WP_Editors::enqueue_scripts(true) #10 /var/www/html/wp-includes/general-template.php(3828): _WP_Editors::enqueue_default_editor() #11 /var/www/html/wp-content/plugins/shortcode-imdb-master/shortcode-imdb.php(17): wp_enqueue_editor() #12 /var/www/html/wp-settings.php(473): include_once('/var/www/html/w...') #13 /var/www/html/wp-config.php(255): require_once('/var/www/html/w...') #14 /var/www/html/wp-load.php(50): require_once('/var/www/html/w...') #15 /var/www/html/wp-admin/admin.php(34): require_once('/var/www/html/w...') #16 /var/www/html/wp-admin/plugins.php(10): require_once('/var/www/html/w...') #17 {main} thrown in /var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.4/script-loader.php on line 38 Notice: Function is_embed was called incorrectly. Conditional query tags do not work before the query is run. Before then, they always return false. Please see Debugging in WordPress for more information. (This message was added in version 3.1.0.) in /var/www/html/wp-includes/functions.php on line 6031 Notice: Function is_search was called incorrectly. Conditional query tags do not work before the query is run. Before then, they always return false. Please see Debugging in WordPress for more information. (This message was added in version 3.1.0.) in /var/www/html/wp-includes/functions.php on line 6031
```

</details>

If we look at [how other plugins are using this function](https://wpdirectory.net/search/01HFMWQQ4T4CA0P6TNX1HDK8J0), it's via hooks, e.g., 

```php
add_action( 'admin_enqueue_scripts', $some_function );
```


```php
add_action( 'wp_enqueue_scripts', $some_function );
```

So, I can confirm that, for the IMDB shortcut plugin, replacing `wp_enqueue_script()` with the following fixes the issue and the plugin works as expected:

```php
add_action( 'wp_enqueue_scripts', function() {
	wp_enqueue_editor();
} );
```

I've added an issue to the [plugin's Github repo](https://github.com/kemalyazici/shortcode-imdb/issues/3)

In short, I'm not sure there's a fix on the Core side.

**Comment by liviopv** [comment]
📌 REPRODUCTION RESULTS
- Tested on Atomic – Replicated

📌 FINDINGS/SCREENSHOTS/VIDEO
Triggered a fatal error when installing the IMDB Shortcode on my Atomic Site:

```PHP Fatal error:  Uncaught TypeError: array_values(): Argument #1 ($array) must be of type array, null given in /wordpress/plugins/gutenberg/17.1.4/lib/compat/wordpress-6.4/script-loader.php:38
Stack trace:
#0 /wordpress/plugins/gutenberg/17.1.4/lib/compat/wordpress-6.4/script-loader.php(38): array_values(NULL)
#1 /wordpress/core/6.4.2/wp-includes/class-wp-hook.php(324): gutenberg_update_wp_date_settings(Object(WP_Scripts))
#2 /wordpress/core/6.4.2/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#3 /wordpress/core/6.4.2/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#4 /wordpress/core/6.4.2/wp-includes/class-wp-scripts.php(166): do_action_ref_array('wp_default_scri...', Array)
#5 /wordpress/core/6.4.2/wp-includes/class-wp-scripts.php(149): WP_Scripts->init()
#6 /wordpress/core/6.4.2/wp-includes/functions.wp-scripts.php(24): WP_Scripts->__construct()
#7 /wordpress/core/6.4.2/wp-includes/functions.wp-scripts.php(368): wp_scripts()
#8 /wordpress/core/6.4.2/wp-includes/class-wp-editor.php(860): wp_enqueue_script('editor')
#9 /wordpress/core/6.4.2/wp-includes/class-wp-editor.php(910): _WP_Editors::enqueue_scripts(true)
#10 /wordpress/core/6.4.2/wp-includes/general-template.php(3828): _WP_Editors::enqueue_default_editor()
#11 /srv/htdocs/wp-content/plugins/shortcode-imdb/shortcode-imdb.php(17): wp_enqueue_editor()
#12 /wordpress/core/6.4.2/wp-settings.php(473): include_once('/srv/htdocs/wp-...')
#13 /srv/htdocs/wp-config.php(81): require_once('/wordpress/core...')
#14 /wordpress/core/6.4.2/wp-load.php(55): require_once('/srv/htdocs/wp-...')
#15 /wordpress/core/6.4.2/wp-blog-header.php(13): require_once('/wordpress/core...')
#16 /wordpress/core/6.4.2/index.php(17): require('/wordpress/core...')
#17 {main}
  thrown in /wordpress/plugins/gutenberg/17.1.4/lib/compat/wordpress-6.4/script-loader.php on line 38```

📌 ACTIONS
- Closed as it's a plugin conflict with core, and reported on the plugin repo https://github.com/kemalyazici/shortcode-imdb/issues/3
 
📌 Message to Author
@villanovachile closing as plugin conflict.
